neo/rawio/axonarawio.py

```diff
@@ -347,35 +347,24 @@ def _get_spike_timestamps(self, block_index, seg_index, unit_index,
         assert block_index == 0
         assert seg_index == 0
 
-        unit_params = self.file_parameters['unit']
-
         tetrode_id = unit_index
         raw_spikes = self._raw_spikes[tetrode_id]
 
         # spike times are repeated for each contact -> use only first contact
         unit_spikes = raw_spikes['spiketimes'][::4]
 
         # slice spike times only if needed
-        if t_start is None or t_stop is None:
-
-            if t_start is None:
-                t_start = self._segment_t_start(block_index, seg_index)
-            if t_stop is None:
-                t_stop = self._segment_t_stop(block_index, seg_index)
+        if t_start is None and t_stop is None:
+            return unit_spikes
 
-            # convert to t_start and t_stop to sampling frequency
-            # Note: this assumes no time offset!
-            lim0 = t_start * self._to_hz(unit_params['timebase'])
-            lim1 = t_stop * self._to_hz(unit_params['timebase'])
+        if t_start is None:
```

Comment:
Are you sure it is save to assume the spiketrains span the complete segment? A segment duration is defined as the time spanned by any of it's contained data. This might lead to earlier start / later stop times for spike trains than actually recorded, e.g. if an additional raw signal was recorded first or setup events were tracked before spike extraction started...

Reply:
I am not 100% sure that is a safe assumption. But it sounds like an edge case when they do not align, most labs will probably not convert only part of the .bin data to unit data and then read both into neo. In the worst case, would it be an issue to have additional noise prior to or post the period with spike data? 

As a more general check of how well unit and raw data align, we could compare the `duration` parameter in the header of a tetrode file (e.g. .1) and the .set file. Given the example data and code we have these will always align however, since the duration is overwritten by their converter. I am not sure if that would also happen if you use the TINT software. 

I am tempted to make the assumption that raw and unit data correspond for now. But I will make a note about it and we can discuss with the Hussaini lab when we present our first prototype. What do you think?

Reply:
Yes, with the current datasets it sounds like a good assumption, so we can keep the current version for now, but potentially need to update this in the future. Keep me updated if you learn something new about this in the context of TINT.